samples/DispatcherSample/Startup.cs

```diff
@@ -1,73 +1,79 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Dispatcher;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Routing;
-using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.AspNetCore.Routing.Dispatcher;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 
 namespace DispatcherSample
 {
     public class Startup
     {
+        private readonly static IInlineConstraintResolver ConstraintResolver = new DefaultInlineConstraintResolver(
+            new OptionsManager<RouteOptions>(
+                new OptionsFactory<RouteOptions>(
+                    Enumerable.Empty<IConfigureOptions<RouteOptions>>(),
+                    Enumerable.Empty<IPostConfigureOptions<RouteOptions>>())));
+
         public void ConfigureServices(IServiceCollection services)
         {
             services.Configure<DispatcherOptions>(options =>
             {
-                options.DispatcherEntryList = new List<DispatcherEntry>()
-                {
-                    new DispatcherEntry
-                    {
-                        RouteTemplate = TemplateParser.Parse("{Endpoint=example}"),
-                        Endpoints = new []
+                options.Dispatchers.Add(CreateDispatcher(
```

Comment:
@jbagga - this is the same set of endpoint and routes you used, I just made it gross-er.